packages/plugin-ext/src/plugin/plugin-manager.ts

```diff
@@ -291,6 +291,14 @@ export class PluginManagerExtImpl implements PluginManagerExt, PluginManager {
                     // see https://github.com/TypeFox/vscode/blob/70b8db24a37fafc77247de7f7cb5bb0195120ed0/src/vs/workbench/api/common/extHostExtensionService.ts#L372-L376
                     pluginMain = pluginMain || {};
                     return await this.startPlugin(plugin, configStorage, pluginMain);
+                } catch (err) {
+                    if (this.pluginActivationPromises.has(plugin.model.id)) {
+                        this.pluginActivationPromises.get(plugin.model.id)!.reject(err);
+                    }
+                    const message = `Activating extension '${plugin.model.displayName || plugin.model.name}' failed: ${err.message}`;
+                    this.messageRegistryProxy.$showMessage(MainMessageType.Error, message, {}, []);
```

Comment:
I was wondering maybe instead of using message service, we should a special service then you can custoimize it in your product to complete stop the execution if some core extensions are failed, by default it will delegate to the notification center.

Reply:
Again, this feels like a separate PR to change architecture rather than this bug fix.

Reply:
ok